src/testUtils/apiServer.ts

```diff
@@ -1,20 +1,259 @@
+import { HTTP_STATUS_CODES } from '../utilities/http.js';
+import { MemberModelSchema, Role } from '../models/Member.model.js';
+import type { Result, SuccessfulResult } from '../utilities/result.js';
 
-import { makeTestServer } from './server.js';
+import { makeTestServer, type TestServerFixture } from './server.js';
 import { OAUTH2_JWKS_URL, OAUTH2_TOKEN_AUDIENCE, OAUTH2_TOKEN_ISSUER } from './authn.js';
-import { configureMockEnvVars, REQUIRED_ENV_VARS } from './envVars.js';
+import { REQUIRED_ENV_VARS } from './envVars.js';
+import { getMockInstance } from './jest.js';
+import { partialPinoLog } from './logging.js';
+import { MEMBER_EMAIL, MEMBER_MONGO_ID, MEMBER_NAME, ORG_NAME } from './stubs.js';
 
-export const REQUIRED_API_ENV_VARS = {
+const ORG_MEMBER: MemberModelSchema = {
+  orgName: ORG_NAME,
+  name: MEMBER_NAME,
+  role: Role.REGULAR,
+  email: MEMBER_EMAIL,
+};
+
+function mockJwksAuthentication(
+  fastify: FastifyInstance,
+  _opts: PluginMetadata,
+  done: PluginDone,
+): void {
+  fastify.decorate(
+    'authenticate',
+    jest.fn<onRequestHookHandler>().mockImplementation((_request, _reply, handlerDone) => {
+      handlerDone();
+    }),
+  );
+
+  done();
+}
+jest.unstable_mockModule('../../utilities/fastify/plugins/jwksAuthentication.js', () => ({
+  default: fastifyPlugin(mockJwksAuthentication, { name: 'mock-jwks-authentication' }),
+}));
+const { makeApiServer } = await import('../api/server.js');
+
+const SUPER_ADMIN_EMAIL = 'admin@veraid-authority.example';
+
+/**
+ * Extract the `authenticate` decorator from the child context of `fastify` where the JWKS plugin
+ * is registered.
+ */
+function getMockAuthenticateFromServer(fastify: FastifyInstance) {
+  const childrenSymbol = Object.getOwnPropertySymbols(fastify).find(
+    (symbol) => symbol.description === 'fastify.children',
+  );
+  if (childrenSymbol === undefined) {
+    throw new Error('Could not find children property');
+  }
+
+  // @ts-expect-error: Allow lookup by symbol
+  const children = fastify[childrenSymbol] as FastifyInstance[];
+
+  const childContext = Object.values(children).find((value) => 'authenticate' in value);
+  if (childContext === undefined) {
+    throw new Error('Could not find child context');
+  }
+  const { authenticate } = childContext;
+  return getMockInstance(authenticate) as jest.Mock<onRequestAsyncHookHandler>;
+}
+
+function setAuthUser(fastify: FastifyInstance, userEmail: string) {
+  // eslint-disable-next-line @typescript-eslint/require-await
+  getMockAuthenticateFromServer(fastify).mockImplementation(async (request) => {
+    (request as unknown as { user: { sub: string } }).user = { sub: userEmail };
+  });
+}
+
+function unsetAuthUser(fastify: FastifyInstance) {
+  getMockAuthenticateFromServer(fastify).mockImplementation(async (_request, reply) => {
+    await reply.code(HTTP_STATUS_CODES.UNAUTHORIZED).send();
+  });
+}
+
+type RouteLevel = 'ORG_BULK' | 'ORG_MEMBERSHIP_RESTRICTED' | 'ORG_MEMBERSHIP' | 'ORG';
```

Comment:
Tried to use an enum first, but because this file has to be imported with `const {RouteLevel} = await import(...)`, that'd break the naming ESLint rule because `RouteLevel` would be a variable and as such it can't be PascalCased.